internal/grafanadi/handler/podsummary_handler.go

```diff
@@ -0,0 +1,211 @@
+package handler
+
+import (
+	"errors"
+	"net/http"
+	"regexp"
+	"sync"
+	"time"
+
+	"github.com/alipay/container-observability-service/internal/grafanadi/model"
+	"github.com/alipay/container-observability-service/pkg/dal/storage-client/data_access"
+	storagemodel "github.com/alipay/container-observability-service/pkg/dal/storage-client/model"
+
+	"github.com/alipay/container-observability-service/pkg/slo"
+	"github.com/alipay/container-observability-service/pkg/utils"
+	interutils "github.com/alipay/container-observability-service/internal/grafanadi/utils"
+	"github.com/alipay/container-observability-service/pkg/metrics"
+	"github.com/alipay/container-observability-service/internal/grafanadi/service"
+)
+
+type PodSummaryHandler struct {
+	request       *http.Request
+	writer        http.ResponseWriter
+	requestParams *PodSummaryParams
+	storage       data_access.StorageInterface
+}
+
+type PodSummaryParams struct {
+	PodUIDName string
+	PodUID     string
+}
+
+func (handler *PodSummaryHandler) RequestParams() interface{} {
+
+	return handler.requestParams
+}
+
+func (handler *PodSummaryHandler) ParseRequest() error {
+	params := PodSummaryParams{}
+	if handler.request.Method == http.MethodGet {
+		params.PodUIDName = handler.request.URL.Query().Get("searchkey")
+		params.PodUID = handler.request.URL.Query().Get("searchvalue")
+	}
+
+	handler.requestParams = &params
+	return nil
+}
+
+func (handler *PodSummaryHandler) ValidRequest() error {
+	return nil
+}
+
+func (handler *PodSummaryHandler) QueryPodSummary(key, value string) (int, interface{}, error) {
+	podInfos := make([]*storagemodel.PodInfo, 0)
+	var podSummary = model.PodSummary{}
+	slotracedata := make([]*storagemodel.SloTraceData, 0)
+	lifephases := make([]*storagemodel.LifePhase, 0)
+	podYamls := make([]*storagemodel.PodYaml, 0)
+	if value == "" {
+		return http.StatusOK, nil, nil
+	}
+
+	begin := time.Now()
+	defer func() {
+		cost := utils.TimeSinceInMilliSeconds(begin)
+		metrics.QueryMethodDurationMilliSeconds.WithLabelValues("QueryPodSummary").Observe(cost)
+	}()
+	util := interutils.Util{
+		Storage: handler.storage,
+	}
+	util.GetUid(podYamls, key, &value)
+	podUid := value
+
+	match, _ := regexp.MatchString("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$", podUid)
+	if !match {
```

Comment:
Is it necessary to reveal the error message to the user?

Reply:
You are right, we can verify the correctness of "podUid" in other parts of the code without returning error messages to the user here. I will delete this code.